hetmech/matrix.py

```diff
@@ -34,6 +34,7 @@ def metaedge_to_adjacency_matrix(graph, metaedge, dtype=numpy.bool_):
         for edge in source_node.edges[metaedge]:
             j = target_node_to_position[edge.target]
             adjacency_matrix[i, j] = 1
+    adjacency_matrix = matrix_type(adjacency_matrix, dtype=dtype)
```

Comment:
It would be ideal to avoid `numpy.zeros` in the sparse case. Doing the conversion after creation still requires the full memory footprint prior to conversion.